js/src/components/experience-rating-banner/index.js

```diff
@@ -0,0 +1,192 @@
+ */
+
+/**
+ * When the user clicks the "Need help" button on the banner.
+ *
+ * @event gla_app_ratings_need_help_clicked
+ * @property {string} context The context in which the event is triggered.
+ */
+
+/**
+ * ExperienceRatingBanner component.
+ *
+ * Displays a dismissible banner asking users to rate their experience with Google for WooCommerce.
+ * Handles user interactions such as clicking "Good", "Need help", and dismissing the banner,
+ * and records corresponding events. Shows a feedback modal when "Good" is clicked.
+ *
+ * @return {JSX.Element|null} The ExperienceRatingBanner component, or null if dismissed.
+ *
+ * @fires gla_app_ratings_shown When the banner is shown.
+ * @fires gla_app_ratings_good_clicked When the "Good" button is clicked.
+ * @fires gla_app_ratings_close When the feedback modal is closed.
+ * @fires gla_app_ratings_need_help_clicked When the "Need help" button is clicked.
+ */
+const ExperienceRatingBanner = () => {
+	const { data: statisticsData } = useMCProductStatistics();
+	const { isReady: isMCAccountReady } = useGoogleMCAccount();
+	const { data: productsReport } = useProductsReport( REPORT_SOURCE_PAID );
+	const [ showModal, setShowModal ] = useState( false );
+	const { set } = useDispatch( preferencesStore );
+	const isDismissed = usePreference( BANNER_DISMISSED_KEY );
+	const statistics = statisticsData?.statistics || {};
+
+	const shouldDisplayBanner = () => {
+		if ( Object.keys( statistics ).length === 0 ) {
+			return false;
+		}
+
+		const totalProducts = Object.values( statistics ).reduce(
+			( total, value ) => total + value,
+			0
+		);
+
+		if ( ! totalProducts ) {
+			return false;
+		}
+
+		const notSyncedProducts = statistics.not_synced;
+		const activeProducts = statistics.active;
+		const conversions = productsReport?.totals?.conversions?.value || 0;
+
+		const hasEnoughActiveProducts =
+			( activeProducts / totalProducts ) * 100 >= 70;
+		const hasAllProductsSynced =
+			notSyncedProducts === 0 && activeProducts > 0;
+		const hasConversions = conversions > 0;
+
+		return (
+			hasEnoughActiveProducts &&
+			hasAllProductsSynced &&
+			isMCAccountReady &&
+			hasConversions
+		);
+	};
+
+	// Fire event when banner is shown.
+	useEffect( () => {
+		if ( ! isDismissed ) {
+			recordGlaEvent( 'gla_app_ratings_shown', {
+				context: APP_RATINGS_BANNER_CONTEXT,
+			} );
+		}
+	}, [ isDismissed ] );
+
+	if ( ! shouldDisplayBanner() ) {
+		return null;
+	}
+
+	const handleGoodOnClick = () => {
+		recordGlaEvent( 'gla_app_ratings_good_clicked', {
+			context: APP_RATINGS_BANNER_CONTEXT,
+		} );
+		setShowModal( true );
+	};
+
+	const handleRequestClose = () => {
+		recordGlaEvent( 'gla_app_ratings_close', {
+			context: APP_RATINGS_BANNER_CONTEXT,
+		} );
+		setShowModal( false );
+	};
+
+	const handleNeedHelpOnClick = () => {
+		recordGlaEvent( 'gla_app_ratings_need_help_clicked', {
+			context: APP_RATINGS_BANNER_CONTEXT,
+		} );
+	};
+
+	const onDismiss = () => {
```

Comment:
Use consistent callback names:
```suggestion
	const handleOnRemove = () => {
```

Reply:
@joemcgill This will be updated in the following ticket https://linear.app/a8c/issue/GOOWOO-198/notification-is-not-dismissed-when-user-clicks-rate-us-button-from-the where we'll rename the function to `dismissBanner` where it'll be used for 2 different click events.